e2e-tests/tests/test_jolteon_consensus.py

```diff
@@ -0,0 +1,180 @@
+from time import sleep, time
+from pytest import mark
+from src.blockchain_api import BlockchainApi, Wallet
+from config.api_config import ApiConfig
+import logging as logger
+
+
+class TestJolteonConsensus:
+    """Test cases for Jolteon consensus protocol implementation (Tachyeon)"""
+    
+    @mark.test_key('JOLTEON-001')
+    def test_qc_formation_and_round_advancement(self, api: BlockchainApi, config: ApiConfig):
+        """Test Jolteon consensus happy path: QC formation and round advancement
+        
+        This test verifies the core Jolteon protocol functionality:
+        - Quorum Certificates (QC) are being formed correctly
+        - Round numbers are advancing as expected
+        - Blocks are being certified with proper consensus
+        
+        Based on Tachyeon test case: Core Protocol Functionality - Happy Path
+        """
+        logger.info("Starting Jolteon consensus QC formation and round advancement test")
+        
+        # Get initial block info
+        initial_block_info = api.substrate.get_block()
+        initial_block_number = initial_block_info['header']['number']
+        initial_round = self._extract_round_number(initial_block_info)
+        
+        logger.info(f"Initial block: {initial_block_number}, round: {initial_round}")
+        
+        # Wait for multiple blocks to be produced and certified
+        # Jolteon typically needs multiple rounds to form QCs and advance
+        wait_time = 30  # Wait 30 seconds to see multiple blocks
+        logger.info(f"Waiting {wait_time} seconds for QC formation and round advancement...")
+        sleep(wait_time)
+        
+        # Get final block info
+        final_block_info = api.substrate.get_block()
+        final_block_number = final_block_info['header']['number']
+        final_round = self._extract_round_number(final_block_info)
```

Comment:
It is great that you have extracted it as a separate method! We can consider moving it to general substrate_api or create a child class of substrate_api with custom consensus methods